ots/timesheet_alias.py

```diff
@@ -65,20 +65,24 @@ def update(self, storage):
         :param odoo: odoorpc.Odoo authenticated to a database
         """
         odoo = storage.load_odoo_session()
+
+        task_id = self.task_id
         task_code = self.task_code
-        if task_code:
+
+        if not task_id and task_code:
```

Comment:
Can you think of a downside if we just leave this first check to be `if task_code:` instead?
The update is built to work mainly based on the task code, and find the task_id based on the task code, but it does not attempt to do it the other way around (update task code based on task id).

So in a hypothetical situation where we have conflicting information at ots end, so a task ID and task code that don't actually match, having the `not task_id` part here would prevent it from correcting itself.

I don't know exactly how we could get into such a situation, but if there is no downside to leaving out the `if not task_id` part, it would probably be better to leave it out.

Since you've now fixed the indentation if the `if task_id` logic branch, the functionality should still stay the same in situations where task code isn't being used at all.